design/m1_push_to_file_design.md

```diff
@@ -0,0 +1,1163 @@
+provides a uniform mechanism for configuring the Secrets Provider, and should
+lend itself to an easier upgrade/migration to the proposed CyberArk Dynamic
+Sidecar Injector design.
+
+## Solution
+
+### User Experience
+
+#### Annotations for Secrets Provider Container/Application Configuration
+
+The Secrets Provider container/application configuration can be set using
+either Pod annotations or by using Pod environment variable settings
+([legacy configuration](https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/11.2/en/Content/Integrations/Kubernetes_deployApplicationsConjur-k8s-Secrets.htm#Step4AddtheCyberArkSecretsProviderforKubernetesimagetotheapplicationdeploymentmanifest)).
+If both a Pod Annotation and a Pod environment variable is configured for
+any particular Secrets Provider setting, then the Pod Annotation configuration
+takes precedence.
+
+_**QUESTION: Should we deprecate the current configuration method that uses
+   environment variable settings?**_
+
+Note that Pod Annotation configuration will not be supported for the
+`MY_POD_NAME` and `MY_POD_NAMESPACE` settings. These settings will continue
+to use the standard
+[Kubernetes Downward API configuration](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/).
+
+The following annotations are supported for Secrets Provider configuration.
+Please refer to the
+[Secrets Provider documentation](https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/11.2/en/Content/Integrations/Kubernetes_deployApplicationsConjur-k8s-Secrets.htm#Step4AddtheCyberArkSecretsProviderforKubernetesimagetotheapplicationdeploymentmanifest)
+for a description of each setting:
+
+| Annotation | Equivalent<br /> Environment Variable | Description, Notes |
+|--------------------------------------|---------------------|----------------------------------|
+| conjur.org/<br />authn-identity      | CONJUR_AUTHN_LOGIN  | Required value. Example: `host/conjur/authn-k8s/cluster/apps/inventory-api` |
+| conjur.org/<br />container-mode      | CONTAINER_MODE      | Allowed values: <ul><li>`init`</li><li>`application`</li><li>`sidecar`</li></ul>Defaults to `sidecar` |
+| conjur.org/<br />secrets-destination | SECRETS_DESTINATION | Required value. Allowed values: <ul><li>`file`</li><li>`k8s_secret`</li></ul> |
+| conjur.org/<br />k8s-secrets         | K8S_SECRETS         | This list is ignored when `conjur.org/secrets-destination` Annotation is set to `file` |
+| conjur.org/<br />retry-count-limit   | RETRY_COUNT_LIMIT   | Defaults to 5
+| conjur.org/<br />retry-interval-sec  | RETRY_INTERVAL_SEC  | Defaults to 1 (sec)              |
+| conjur.org/<br />debug-logging       | DEBUG               | Defaults to `false`              |
+| conjur.org/<br />conjur-configmap    | CONJUR_AUTHN_URL<br />CONJUR_URL<br />CONJUR_ACCOUNT<br />CONJUR_SSL_CERTIFICATE | ConfigMap containing Conjur connection information. |
```

Comment:
will we have a default value for the configmap name? I think that'd be nice to have - then people could skip it if they're using the same name.

also, do we assume the configmap is in the same namespace as the pod? can we be clear about that here?

Reply:
maybe use `path` instead of `dir` for consistency? sounds good in any case; comment about defaults still applies